packages/blockchain-extension/extension/commands/packageSmartContractCommand.ts

```diff
@@ -216,15 +220,77 @@ async function packageJsonNameAndVersion(workspaceDir: vscode.WorkspaceFolder, o
         const message: string = 'Please enter a package name and/or package version into your package.json';
         throw new Error(message);
     }
+
+    const dependencies: any = workspacePackageObj.dependencies;
+    contractAPI = dependencies['fabric-contract-api'];
+
+    if (!contractAPI) {
+        // check if low level chaincode
+        contractAPI = dependencies['fabric-shim'];
+    }
+
+    const contractSemver: semver.SemVer = semver.coerce(contractAPI); // Contract semver
+
+    const contractMajor: number = contractSemver.major;
+
+    // Check if fabric contract API version is greater than 2.0.0
+    if (contractMajor !== 2) {
+        // Version is not greater than or equal to 2
+        throw new Error(`Unable to package contract. Contract API dependency must support Fabric 2. Your version: ${contractAPI}`);
+    }
+
     return { workspacePackageName, workspacePackageVersion };
 }
 
 /**
  * Method which calls an input box should the project be coded in java, which asks the user for a package name and version
  * (as java projects do not contain a package.json file), and returns an object containing both these values.
+ * @param workspaceDir {String} workspaceDir A string containing the path to the current active workspace (the workspace of the project the user is packaging).
  * @returns {string, string} Returns an object with the workspacePackageName and workspacePackageVersion which will be used in the createPackageDir() method
  */
-async function javaPackageAndVersion(overrideName?: string, overrideVersion?: string): Promise<{ workspacePackageName: string, workspacePackageVersion: string }> {
+async function javaPackageAndVersion(workspaceDir: vscode.WorkspaceFolder, overrideName?: string, overrideVersion?: string): Promise<{ workspacePackageName: string, workspacePackageVersion: string }> {
+
+    let contractVersion: string = 'fabric-chaincode-shim';
+    let workspaceGradleMaven: string = path.join(workspaceDir.uri.fsPath, 'build.gradle');
+
+    const exists: boolean = await fs.pathExists(workspaceGradleMaven);
+    if (!exists) {
+        workspaceGradleMaven = path.join(workspaceDir.uri.fsPath, 'pom.xml');
+        const pomExists: boolean = await fs.pathExists(workspaceGradleMaven);
+        if (!pomExists) {
+            throw new Error('Unable to determine contract API version - no build.gradle or pom.xml found');
+        }
+        contractVersion = 'fabric-chaincode-java';
```

Comment:
If the gradle file doesn't exist but the pom file does, are we supposed to only search for fabric-chaincode-java?